doc/zettelkasten.txt

```diff
@@ -86,11 +86,15 @@ COMMANDS                                                  *zettelkasten-commands
 
 Global ~
 
-ZkNew                                                                    *:ZkNew*
+ZkNew {parent_id}                                                        *:ZkNew* 
 
     Creates a new buffer with a markdown file type. If {notes_path} option was
     set with |zettelkasten-setup|, it'll change the working directory.
 
+    Parameters: ~
+        {parent_id} (optional, string): the parent id to use when choosing the
```

Comment:
It's not clear to me what this `parent_id` is and why it is needed.

Reply:
So, the idea is to provide the `id_format` function some context on where it is generating an id. For the current use of generating ids from timestamps this is irrelevant.  But if one (like me) wanted to use Ludmann's naming scheme where notes are numbered sequentially: 1,2,3 but can have child notes inserted 1, 1a, 1b, 2, 3, then knowing the `parent_id` is helpful for generating new ids. Namely if one wanted to add to 1a, 1b, sequence one would call `:ZkNew 1`.

I have expanded the documentation a bit to try and explain this in a more generic way.

It is also an optional argument so does not change the public API, but it does "pollute" it a bit. 
What do you think?